app/src/main/java/org/oppia/android/app/hintsandsolution/HintsAndSolutionDialogFragmentPresenter.kt

```diff
@@ -211,7 +211,13 @@ class HintsAndSolutionDialogFragmentPresenter @Inject constructor(
     }
 
     binding.hintTitle.text =
-      resourceHandler.capitalizeForHumans(hintsViewModel.title.get()!!.replace("_", " "))
+      resourceHandler.capitalizeForHumans(
+        resourceHandler.getStringInLocaleWithWrapping(
+          R.string.hint,
+          hintsViewModel.title.get()!!
```

Comment:
Would probably be cleaner to get rid of the title here and just compute the index property in-line.

Reply:
This logic has been moved to the ViewModel and the index is computed and observed directly from the viewmodel.